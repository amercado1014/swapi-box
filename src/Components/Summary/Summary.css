.summary {
  animation: height 22s linear;
  color: black;
  display: flex;
  height: 0px;
  justify-content: center;
  margin: 0 auto;
  padding: 10px;
  perspective: 400px;
  overflow: hidden;
  width: 450px;
}

.film {
  animation: crawl 120s linear;
  transform-origin: 50% 100%;
  position: relative;
  color: #feda4a;
  font-size: 30px;
  text-align: center;
}

.film-text {
  text-align: justify;
}


@keyframes crawl {
  0% {
    top: 0;
    transform: rotateX(20deg) translateZ(0);
  }
  100% {
    top: -6000px;
    min-height: 0px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}

@keyframes height {
  0% {
    height: 300px;
  }
  100% {
    height: 300px;
  }
}

@media (max-width: 500px) {
  .summary {
    animation: height 25s linear;
    width: 300px;
  }
}
