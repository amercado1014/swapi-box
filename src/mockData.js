const mockCardData = {
  name: "Luke Skywalker",
  class: "people-card",
  data: {
    homeworld: "Tatooine",
    language: "Galactic Basic",
    population: "200000",
    species: "Human"
  }
};

const mockCardContainerData = [
  {
    name: "Luke Skywalker",
    class: "people-card",
    data: {
      homeworld: "Tatooine",
      language: "Galactic Basic",
      population: "200000",
      species: "Human"
    }
  },
  {
    name: "Darth Vader",
    class: "people-card",
    data: {
      homeworld: "Tatooine",
      language: "Galactic Basic",
      population: "200000",
      species: "Human"
    }
  },
  {
    name: "C-3PO",
    class: "people-card",
    data: {
      homeworld: "Tatooine",
      language: "n/a",
      population: "200000",
      species: "Droid"
    }
  }
];

const mockSummaryData = {
  date: "1983-05-25",
  episode: 6,
  text: "Luke Skywalker has returned to\r\nhis home planet of Tatooine",
  title: "Return of the Jedi"
};

export {
  mockCardData,
  mockCardContainerData,
  mockSummaryData
};