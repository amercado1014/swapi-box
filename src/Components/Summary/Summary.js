import React from 'react';
import './Summary.css';

const Summary = ({film}) => {
  return (
    <div className="summary">
      <div className="fade">
        <section className="star-wars">
          <div className="crawl">
            <div className="title">
              <p>Episode {film.episode}</p>
              <h1>{film.title}</h1>
            </div>
            <p>{film.text}</p>
          </div>
        </section>
      </div>
    </div>
  );
};

export default Summary;