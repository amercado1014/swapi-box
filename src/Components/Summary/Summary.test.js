import React from 'react';
import Summary from './Summary';
import { shallow } from 'enzyme';
import { mockSummaryData } from '../../mockData';

describe('Summary', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Summary film={mockSummaryData} />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});