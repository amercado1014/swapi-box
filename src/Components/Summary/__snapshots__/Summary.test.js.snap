// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Summary
    film={
        Object {
            "date": "1983-05-25",
            "episode": 6,
            "text": "Luke Skywalker has returned to
        his home planet of Tatooine",
            "title": "Return of the Jedi",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="fade"
>
        <section
                className="star-wars"
        >
                <div
                        className="crawl"
                >
                        <div
                                className="title"
                        >
                                <p>
                                        Episode 
                                        6
                                </p>
                                <h1>
                                        Return of the Jedi
                                </h1>
                        </div>
                        <p>
                                Luke Skywalker has returned to
                                his home planet of Tatooine
                        </p>
                </div>
        </section>
</div>,
      "className": "summary",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="star-wars"
>
          <div
                    className="crawl"
          >
                    <div
                              className="title"
                    >
                              <p>
                                        Episode 
                                        6
                              </p>
                              <h1>
                                        Return of the Jedi
                              </h1>
                    </div>
                    <p>
                              Luke Skywalker has returned to
                              his home planet of Tatooine
                    </p>
          </div>
</section>,
        "className": "fade",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
>
            <div
                        className="title"
            >
                        <p>
                                    Episode 
                                    6
                        </p>
                        <h1>
                                    Return of the Jedi
                        </h1>
            </div>
            <p>
                        Luke Skywalker has returned to
                        his home planet of Tatooine
            </p>
</div>,
          "className": "star-wars",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                <p>
                                Episode 
                                6
                </p>
                <h1>
                                Return of the Jedi
                </h1>
</div>,
              <p>
                Luke Skywalker has returned to
                his home planet of Tatooine
</p>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Episode 
                    6
</p>,
                  <h1>
                    Return of the Jedi
</h1>,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Episode ",
                      6,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Episode ",
                    6,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Return of the Jedi",
                  },
                  "ref": null,
                  "rendered": "Return of the Jedi",
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Luke Skywalker has returned to
his home planet of Tatooine",
              },
              "ref": null,
              "rendered": "Luke Skywalker has returned to
his home planet of Tatooine",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "section",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="fade"
>
          <section
                    className="star-wars"
          >
                    <div
                              className="crawl"
                    >
                              <div
                                        className="title"
                              >
                                        <p>
                                                  Episode 
                                                  6
                                        </p>
                                        <h1>
                                                  Return of the Jedi
                                        </h1>
                              </div>
                              <p>
                                        Luke Skywalker has returned to
                                        his home planet of Tatooine
                              </p>
                    </div>
          </section>
</div>,
        "className": "summary",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <section
            className="star-wars"
>
            <div
                        className="crawl"
            >
                        <div
                                    className="title"
                        >
                                    <p>
                                                Episode 
                                                6
                                    </p>
                                    <h1>
                                                Return of the Jedi
                                    </h1>
                        </div>
                        <p>
                                    Luke Skywalker has returned to
                                    his home planet of Tatooine
                        </p>
            </div>
</section>,
          "className": "fade",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="crawl"
>
              <div
                            className="title"
              >
                            <p>
                                          Episode 
                                          6
                            </p>
                            <h1>
                                          Return of the Jedi
                            </h1>
              </div>
              <p>
                            Luke Skywalker has returned to
                            his home planet of Tatooine
              </p>
</div>,
            "className": "star-wars",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
>
                  <p>
                                    Episode 
                                    6
                  </p>
                  <h1>
                                    Return of the Jedi
                  </h1>
</div>,
                <p>
                  Luke Skywalker has returned to
                  his home planet of Tatooine
</p>,
              ],
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      Episode 
                      6
</p>,
                    <h1>
                      Return of the Jedi
</h1>,
                  ],
                  "className": "title",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Episode ",
                        6,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Episode ",
                      6,
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Return of the Jedi",
                    },
                    "ref": null,
                    "rendered": "Return of the Jedi",
                    "type": "h1",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Luke Skywalker has returned to
his home planet of Tatooine",
                },
                "ref": null,
                "rendered": "Luke Skywalker has returned to
his home planet of Tatooine",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "section",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
